### PR TITLE
Extract `ExecutionContext` to its own module and cli code to `CliApp`

### PR DESCRIPTION
**Pull request by alamb**
Part of https://github.com/datafusion-contrib/datafusion-tui/issues/132

This PR extracts the ExecutionContext into its own module, as a first step towards
adding additional features (e.g. other table formats)

Having it as its own module will allow us to test it more easily and keeps the boundaries of the software clearer.

Changes:
1. Move ExecutionContext to its own module
2. Make fields not `pub`
3. Encapsulate the CLI code into its own module

### CODE REVIEW

File: src/app/handlers/flightsql.rs [hunk 1]
```diff
@@ -73,7 +73,7 @@ pub fn normal_mode_handler(app: &mut App, key: KeyEvent) {
             let execution = Arc::clone(&app.execution);
             let _event_tx = app.app_event_tx.clone();
             tokio::spawn(async move {
-                let client = &execution.flightsql_client;
+                let client = execution.flightsql_client();
```
Comment (alamb) [inline src/app/handlers/flightsql.rs:76]: I just made the fields non pub and made accessors so the API surface area was clearer

File: src/app/mod.rs [hunk 2]
```diff
@@ -328,84 +327,98 @@ pub async fn run_app(cli: cli::DftCli, state: state::AppState<'_>) -> Result<()>
...
-) -> Result<()> {
-    match (files.is_empty(), commands.is_empty()) {
-        (true, true) => Err(eyre!("No files or commands provided to execute")),
-        (false, true) => execute_files(files, state).await,
-        (true, false) => execute_commands(commands, state).await,
-        (false, false) => Err(eyre!(
-            "Cannot execute both files and commands at the same time"
-        )),
-    }
+/// Encapsulates the command line interface
```
Comment (alamb) [inline src/app/mod.rs:330]: I refactored a bunch of free functions into a struct (which I plan to move to its own module soon)

File: src/execution/mod.rs [hunk 3]
```diff
@@ -40,21 +41,34 @@ use {
...
+///
+/// 2. The code for running SQL queries
+///
+/// The design goals for this module are to serve as an example of how to integrate
+/// DataFusion into an application and to provide a simple interface for running SQL queries
+/// with the various extensions enabled.
+///
+/// Thus it is important (eventually) not depend on the code in the app crate
 pub struct ExecutionContext {
-    pub session_ctx: SessionContext,
```
Comment (alamb) [inline src/execution/mod.rs:48]: when I made these non pub it turned out much of it isn't used, so I removed the unused fields for now

File: src/execution/mod.rs [hunk 4]
```diff
@@ -40,21 +41,34 @@ use {
...
 };
 
-use super::config::ExecutionConfig;
-use super::state::tabs::sql::Query;
-use super::AppEvent;
+use crate::app::config::ExecutionConfig;
+use crate::app::state::tabs::sql::Query;
+use crate::app::AppEvent;
 
+/// Structure for executing queries either locally or remotely (via FlightSQL)
```
Comment (alamb) [inline src/execution/mod.rs:48]: This is the core reason for this PR -- to move this structure to its own module

It still depends on `app::AppEvent` as well as printing to stdout. I plan to move the display of results code into different structures in the next PR

**Review by alamb — COMMENTED** [review]
FYI @matthewmturner  this is step one of a few PRs I have planned. I filed https://github.com/datafusion-contrib/datafusion-tui/issues/132 with my rough plan of attack

File: src/app/handlers/flightsql.rs (hunk 1 above)
Comment (matthewmturner) [reply src/app/handlers/flightsql.rs:76]: Yup makes sense, i had started to use that approach for more recent work but i know theres some remaining pub stuff

**Review by matthewmturner — APPROVED** [review]

**Pull request merged**